Sources/TuistLoader/Models+ManifestMappers/Destination+ManifestMapper.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+import ProjectDescription
+import TuistGraph
+
+extension TuistGraph.Destination {
+    /// Maps a ProjectDescription.Package instance into a TuistGraph.Package model.
+    /// - Parameters:
+    ///   - manifest: Manifest representation of Package.
+    ///   - generatorPaths: Generator paths.
+    static func from(
+        platform: ProjectDescription.Platform,
+        deploymentTarget: ProjectDescription.DeploymentTarget?
+    ) throws -> TuistGraph.Destinations {
+        switch (platform, deploymentTarget) {
+        case (.macOS, _):
+            return [.mac]
+        case let (.iOS, .some(.iOS(_, devices, supportsMacDesignedForIOS: supportsMacDesignedForIOS))):
+            var destinations: [Destination] = []
+
+            if devices.contains(.iphone) {
+                destinations.append(.iPhone)
+            }
+
+            if devices.contains(.ipad) {
+                destinations.append(.iPad)
+            }
+
+            if devices.contains(.mac) {
+                destinations.append(.macCatalyst)
+            }
+
+            if devices.contains(.vision) {
+                destinations.append(.appleVisionWithiPadDesign)
+            }
+
+            if supportsMacDesignedForIOS {
+                destinations.append(.macWithiPadDesign)
+            }
+
+            return Set(destinations)
+        case (.iOS, _): // an iOS platform, but `nil` deployment target.
+            return [.iPhone, .iPad, .macCatalyst, .appleVisionWithiPadDesign]
```

Comment:
Is specifying `.macCatalyst` correct here?

Should it be this instead?

```suggestion
            return .iOS
```

Which is currently defined as:

```swift
public static var iOS: Destinations = [.iPhone, .iPad, .macWithiPadDesign]
```


Adding `.macCatalyst` here would mean any project that doesn't specify a deployment target would get a `.macCatalyst` destination.

For example, the `App` target in the `ios_app_with_tests` fixture 

Latest Tuist:

![latest](https://github.com/tuist/tuist/assets/11914919/e642cfa9-0fa7-4185-981d-52d4362a787a)


vs this PR:

![pr](https://github.com/tuist/tuist/assets/11914919/a5e9477c-7db0-435d-885d-c3f671b1f923)